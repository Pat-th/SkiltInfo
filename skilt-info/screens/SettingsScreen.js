import React, {useEffect, useState} from 'react';
import {AsyncStorage, FlatList, Image, StyleSheet, Text, TouchableOpacity, View, Alert, TextInput, Picker} from 'react-native';
import Collapsed from "../components/Collapsed";
import SettingsFilters from "../components/SettingsFilters";
import Colors  from "../Constants/Colors"


const SettingsScreen = props => {
    let filters = ['hei', 'ho'];
    const [data, setData] = useState([]);
    const [selected, setSelected] = useState('Enkel');
    const [radius, setRadius] = useState('500');
    const [selectedSign, setSelectedSign] = useState("7644");


    useEffect(() => {
        getData();
    }, [filters]);

    useEffect(()=> {
        setStates();
    },[]);

    const setStates = async () => {
        let distance = await AsyncStorage.getItem('radius');
        setRadius(distance);
        let signType = await AsyncStorage.getItem('signType');
        setSelectedSign(signType);
    };

    const getData = async () => {
        filters = [];
        let getFilters = await AsyncStorage.getItem('filters');
        let result = await JSON.parse(getFilters);
        if(result.filters.length > 3) {
            for (let i = 3; i < result.filters.length; i++) {
                filters.push(
                    result.filters[i]
                );
            }
        }
        let standard = await AsyncStorage.getItem('standard');
        await setSelected(standard);
        setData(filters);
    };



    const setFilter = async filter => {
        await AsyncStorage.setItem('standard', filter);
        setSelected(filter);
    };

    const goToAddNew = () => {
        props.navigation.navigate('Nytt Filter')
    };

    const deleteAlert = filterGettingDeleted => {
        Alert.alert(
            'Sletting av filter',
            'Er du sikker du vil slette dette filteret?'    ,
            [{text: 'Avbryt', style: 'cancel'},
                {text: 'SLETT', onPress: () => deleteFilter(filterGettingDeleted)}]
        )

    };

    const deleteFilter = async (filterGettingDeleted) => {
        filters = [];
        const getFilters = await AsyncStorage.getItem('filters');
        let parseFilters = await JSON.parse(getFilters);

        let index = parseFilters.filters.indexOf(filterGettingDeleted);

        if (index > -1) {
            parseFilters.filters.splice(index, 1);
        }
        let stringifyFilters = JSON.stringify(parseFilters);
        await AsyncStorage.setItem('filters', stringifyFilters);
        await AsyncStorage.removeItem(filterGettingDeleted);
        for (let i = 3; i < parseFilters.filters.length; i++) {
            filters.push(
                parseFilters.filters[i]
            );
        }
        if(filterGettingDeleted === selected){
            await setFilter('Enkel')
        }
        setData(filters);
    };

    const saveRadius = async () => {
      await AsyncStorage.setItem('radius', radius);
        let storagedRadius = await AsyncStorage.getItem('radius');
        Alert.alert(
          'Du har gjort en endring!',
          'Du har endret radius til: ' + storagedRadius,
          [{text: 'OK', onPress: () => console.log(storagedRadius)}]

      )
    };

    const saveSignType = async signValue => {
        setSelectedSign(signValue);
        console.log(signValue);
        await AsyncStorage.setItem('signType', signValue, (err, res) => console.log(res));
    };

    const onChangeTextInput = (text) => {
        if (/^\d+$/.test(text) || text === '') {
            setRadius(text);
        }
    };

    const fillArray = () => {
        return (
            <View>
                <View style={styles.textbox}><TouchableOpacity style={{backgroundColor: selected === 'Enkel' ? Colors.accentColor1 : 'transparent', borderRadius: 10,}}
                                        onPress={() => setFilter('Enkel')}><Text
                    style={styles.content}>Enkel</Text></TouchableOpacity></View>
                <View style={styles.textbox}><TouchableOpacity style={{backgroundColor: selected === 'Avansert' ? Colors.accentColor1 : 'transparent', borderRadius: 10,}}
                                        onPress={() => setFilter('Avansert')}><Text
                    style={styles.content}>Avansert</Text></TouchableOpacity></View>
                <View style={styles.textbox}><TouchableOpacity style={{backgroundColor: selected === 'Fullstendig' ? Colors.accentColor1 : 'transparent', borderRadius: 10,}}
                                        onPress={() => setFilter('Fullstendig')}><Text
                    style={styles.content}>Fullstendig</Text></TouchableOpacity></View>
                <FlatList
                    data={data}
                    keyExtractor={(item, index) => index.toString()}
                    renderItem={({item}) => {
                        return (
                            <SettingsFilters
                                selectorStyle={{backgroundColor: selected === item ? Colors.accentColor1 : 'transparent', borderRadius: 10}}
                                selector={() => setFilter(item)}
                                textStyle={styles.content}
                                deleteButton={() => deleteAlert(item)}
                                sendToEdit={item}
                                text={item}
                                editButton={() => props.navigation.navigate('Rediger Filter', {toEdit: item})}
                                extraData={selected}
                            />)
                    }}
                />
                <View style={styles.textbox}>
                    <TouchableOpacity onPress={() => goToAddNew()}>
                        <Text style={styles.content}>Legg til nytt filter</Text>
                        <Image source={require('../images/plus.png')} style={styles.plusicon}/>
                    </TouchableOpacity>
                </View>
            </View>)
    };

    return (
        <View style={styles.bigContainer}>
            <Collapsed
                TitleStyle={styles.headerText}
                titleOfCollapsible={"Filter"}
                contentOfCollapsible={fillArray()}
                viewStyle={{padding: 5}}
            />
            <View style={styles.container}>
                <Text style={styles.headerText}>Radius</Text>
                <View style={styles.radiusgroup}>
                    <TextInput
                    underlineColorAndroid='transparent'
                    keyboardType={'numeric'}
                    value={radius.toString()}
                    style={styles.radiusinput}
                    onChangeText={onChangeTextInput}/>
                    <TouchableOpacity style={styles.save} onPress={() => saveRadius()}><Image source={require('../images/save.png')} style={styles.saveicon}/></TouchableOpacity>
                </View>
            </View>
            <View style={styles.container}>
                <Text style={styles.headerText}>Skilttype</Text>
                <Picker
                    selectedValue={selectedSign}
                    style={{ height: 50, width: 150, position: "absolute", right: 10 }}
                    onValueChange={(signValue) => saveSignType(signValue)}
                >
                    <Picker.Item label="ForkjÃ¸rsveg" value="7644" />
                    <Picker.Item label="Vikeplikt" value="7642" />
                    <Picker.Item label="Gangfelt" value="7705" />
                </Picker>
            </View>
        </View>
    )
};

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        padding: 5,
    },
    content: {
        padding: 5,
        textAlign: 'left',
    },
    headerText: {
        textAlign: 'left',
        fontSize: 24,
        fontWeight: '500',
        color: 'rgba(0,0,0,1)',
    },
    switchbutton: {
        width: 20,
        height: 15,
        position: 'absolute',
        right: 20,
    },
    plusicon: {
        height: 20,
        width: 20,
        position: 'absolute',
        right: 10,
        marginTop: 4
    },
    saveicon: {
        height: 25,
        width: 25,
        marginTop: 3
    },
    save: {
        margin: 10,
        height: 30,
        width: 50,
        alignItems: 'center'
    },
    textbox: {
        margin: 3,
        borderWidth: 1,
        borderRadius: 10,
        justifyContent: 'center',
    },
    radiusgroup: {
        position: 'absolute',
        right: 10,
        flexDirection: 'row',
        alignItems: 'center',
    },
    radiusinput: {
        fontSize: 24,
        width: 60,
        borderBottomColor: "#a38e8c",
        borderBottomWidth: 0.8,
        textAlign: 'center',
        height: 30,
    },
    bigContainer: {
        margin: 10
    }
});

export default SettingsScreen;