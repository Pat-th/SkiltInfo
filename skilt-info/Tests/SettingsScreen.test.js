import React from "react";
import renderer from "react-test-renderer";

import SettingsScreen from "../screens/SettingsScreen";

describe("<SettingsScreen />", () => {
    it("Has 1 child", () => {
        const tree = renderer.create(<SettingsScreen />).toJSON();
        expect(tree.children.length).toBe(1);
    });
    it('Renders correctly', () => {
        const tree = renderer.create(<SettingsScreen />).toJSON();
        expect(tree).toMatchSnapshot();
      });
   });