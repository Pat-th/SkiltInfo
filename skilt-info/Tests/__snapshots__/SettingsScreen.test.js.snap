// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> Renders correctly 1`] = `
<View>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(0,0,0,1)",
            "fontSize": 24,
            "fontWeight": "500",
            "textAlign": "left",
          }
        }
      >
        Filter
      </Text>
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "height": 0,
          "overflow": "hidden",
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "justifyContent": "center",
            "padding": 10,
          }
        }
      >
        <View>
          <View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#6e3b6e",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgba(0,0,0,1)",
                    "padding": 5,
                    "textAlign": "left",
                  }
                }
              >
                Enkel
              </Text>
            </View>
          </View>
          <View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#f9c2ff",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgba(0,0,0,1)",
                    "padding": 5,
                    "textAlign": "left",
                  }
                }
              >
                Avansert
              </Text>
            </View>
          </View>
          <View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "#f9c2ff",
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "rgba(0,0,0,1)",
                    "padding": 5,
                    "textAlign": "left",
                  }
                }
              >
                Fullstendig
              </Text>
            </View>
          </View>
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
          <View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text>
                Legg til nytt filter
              </Text>
              <Image
                source={1}
                style={
                  Object {
                    "height": 25,
                    "position": "absolute",
                    "right": 10,
                    "width": 25,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(0,0,0,1)",
          "fontSize": 24,
          "fontWeight": "500",
          "textAlign": "left",
        }
      }
    >
      Nattmodus
    </Text>
    <RCTSwitch
      accessibilityRole="button"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "height": 15,
            "position": "absolute",
            "right": 20,
            "width": 20,
          },
        ]
      }
      value={false}
    />
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        fetch
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        slett alt
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        selected
      </Text>
    </View>
  </View>
</View>
`;
